### ISSUE TITLE
GTK3 Nightly Builds

### ISSUE THREAD

**Issue by elextr** [geany/geany#2003]
Since many distros now distribute the GTK3 version of Geany the nightly builds should also check GTK3.

**Comment by eht16** [comment]
Alternatively, we could even remove the GTK2 nightly builds as they are pretty much obsoleted by the Travis builds which are run more often and even against PRs.
In contrary to the Debian nightly builds, the build results are of no further use and so are no better than the Travis builds, IMO.

I see only one advantage over Travis: if we would build the nightlies against the minimum required versions of GTK2 and GTK3 to ensure compatibility. But this would require compiling the whole GTK2 and GTK3 stack manually to compile against (the current GTK2 nightlies are built against the GTK2 stack provided by Debian Stable).

**Comment by elextr** [comment]
Yeah, I saw the benefit of the nightlys that they could use different versions of GTK from Travis, but compiling the whole stack seems a bit OTT.

But otherwise they don't add a lot any more.

**Comment by eht16** [comment]
Hence the proposal to drop the GTK2 builds instead of adding new ones.

OTOH, compiling the whole stack for GTK 2.24 and GTK 3.0 probably isn't the hardest thing ever, it just needs to be done. And it needs to be done only once we change the minimum required versions or a new Debian version is released (because of dist-upgrade and binary incompatibilities). The rebuilding can probably be solved easily scripted.

**Comment by codebrainz** [comment]
I believe people actually use the Windows nightly binaries, but probably not much on other OS.

**Comment by eht16** [comment]
> I believe people actually use the Windows nightly binaries, but probably not much on other OS.

I'd be curious how. As mentioned in
https://github.com/geany/geany/issues/1979, the current Windows nightly
builds are not compatible with the usual releases since the MSYS2 switch.

@b4n is it worth the effort and worth the increased build time on Travis? This would grow the build matrix. I think having test builds against the minimum GTK versions is nice to have but probably enough once a day and not for each push.

**Comment by codebrainz** [comment]
> I'd be curious how.

Well I meant we've gotten bug reports when it doesn't work, at least previously before the statement about using them was removed since they aren't compatible. FWIW, I don't think it was a good way to test bleeding edge Geany. In the future if/when anyone gets motivated to do it, it would be better to have the whole installer(s) built automatically when master branch is updated.

**Comment by elextr** [comment]
There are no issues about the nightly builds not being runable for several years, and the last ones by that @eht16 guy so they don't count :grin:  So I would say that its a non-issue.

Travis builds the pull requests, it doesn't build master after merges, thats one thing nightly builds can do that Travis doesn't.  That will pick up any issues with one PR affecting another even if they have no conflicts in Git.

As @codebrainz said, it should build an installable version for windows, and maybe a generic Linux binary tarball to encourage testing (ok, thats hopeful but still :)

**Comment by eht16** [comment]
> Travis builds the pull requests, it doesn't build master after merges, thats one thing nightly builds can do that Travis doesn't.  That will pick up any issues with one PR affecting another even if they have no conflicts in Git.

I think Travis *does* build also on pushes to master, e.g. https://travis-ci.org/geany/geany/builds/464017469 this was a regular commit without merge or PR, I think.

> As @codebrainz said, it should build an installable version for windows, and maybe a generic Linux binary tarball to encourage testing (ok, thats hopeful but still :)

I agree, at least for the Windows installer part. Providing a Linux binary would be easy but I'm not sure how generic it can be built since we can't link statically. Maybe we could Flatpak or so for this if at all.
The Windows installer generation is more fun: luckily NSIS exists for Linux and so cross-compiling the installer should be possible. The problem is just to get a MSYS2 environment running on Linux for the dependency management. In theory it could work with Wine but I never tested it.

**Comment by eht16** [comment]
I'd like to close this.
We do have GTK3 nightly and CI builds and there is no more Travis, six years later :).
